chrome-launcher/chrome-launcher.ts

```diff
@@ -45,24 +48,79 @@ export interface ModuleOverrides {
   spawn?: typeof childProcess.spawn;
 }
 
-export async function launch(opts: Options = {}): Promise<LaunchedChrome> {
-  opts.handleSIGINT = defaults(opts.handleSIGINT, true);
+/**
+ * LaunchManager handles launch retries and SIGINT binding
+ */
+export class LaunchManager {
+  private instance?: Launcher;
+  private opts: Options;
+  private isSigintBound: boolean;
+  private maxLaunchAttempts: number;
+  private attempts: number;
+
+  constructor() {
+    this.isSigintBound = false;
+    this.maxLaunchAttempts = 3;
+    this.attempts = 1;
+  }
+
+  setOptions(opts: Options = {}) {
+    this.opts = opts;
+  }
+
+  async launchInstance(): Promise<LaunchedChrome> {
+    if (this.instance) {
+      Promise.reject(new Error('Already a launcher instance, can\'t create a second.'));
+    }
 
-  const instance = new Launcher(opts);
+    this.instance = new Launcher(this.opts);
 
-  // Kill spawned Chrome process in case of ctrl-C.
-  if (opts.handleSIGINT) {
-    process.on(_SIGINT, async () => {
-      await instance.kill();
-      process.exit(_SIGINT_EXIT_CODE);
+    this.handleSigint();
+    await this.instance.launch().catch(err => {
+      return this.retry(err);
     });
+
+    return {
+      pid: this.instance.pid!,
+      port: this.instance.port!,
+      kill: async () => this.instance!.kill()
+    };
   }
 
-  await instance.launch();
+  async retry(err: LighthouseError) {
+    if (err.code !== 'ECONNREFUSED') {
+      Promise.reject(err);
+    }
+    this.attempts++;
+    if (this.attempts <= this.maxLaunchAttempts) {
+      log.warn(
+          'ChromeLauncher',
+          `Connection refused. Retrying attempt #${this.attempts}/${this.maxLaunchAttempts}`);
+    } else {
+      log.error('ChromeLauncher', 'Reached maximum relaunch attempts. Quitting...');
+      return Promise.reject(err);
+    }
+
+    await this.instance!.kill();
+    this.instance = undefined;
+    return this.launchInstance();
+  }
 
-  return {pid: instance.pid!, port: instance.port!, kill: async () => instance.kill()};
+  async handleSigint() {
+    this.opts.handleSIGINT = defaults(this.opts.handleSIGINT, true);
+    // Kill spawned Chrome process in case of ctrl-C.
+    if (this.opts.handleSIGINT && !this.isSigintBound) {
+      this.isSigintBound = true;
+      process.on(_SIGINT, async () => {
```

Comment:
you mean being async in handling these? Yeah tested it a few times, it works well.

Also didn't mention i got rid of `process.exit()` in launcher as it was technically unnecessary.